### PR TITLE
Cost matrix early abort

### PR DESCRIPTION
**Pull request by monteirodom**
Shorten down the request delay, when some sources/targets searches are early aborted.
* Decrease remaining_sources_ for all targets when a source search is aborted due to overpassing the threshold
* Decrease remaining_targets_ for all sources when a target search is aborted due to overpassing the threshold

## Issue
#3610

### CODE REVIEW

**Comment by monteirodom** [comment]
Yeah, I've seen the tests failing ... no idea why the changes make some good results to be broken. Some help would be appreciated on running the tests: tried to have the tests built but with no success till now (on macOS).
Tried on docker, but failed also at 74% of compilation. Here's the last lines of the compilation
```
#14 2337.1 [ 72%] Linking CXX executable gurka_route_summary
#14 2350.2 [ 72%] Built target gurka_route_summary
#14 2350.3 [ 72%] Generating scripts.log
#14 2350.4 + /usr/bin/python3.8 -m unittest discover -s /src/valhalla/test/scripts -v
#14 2359.9 make[3]: *** [test/scripts/CMakeFiles/run-scripts.dir/build.make:75: test/scripts/scripts.log] Error 1
#14 2360.1 make[2]: *** [CMakeFiles/Makefile2:16848: test/scripts/CMakeFiles/run-scripts.dir/all] Error 2
#14 2360.1 make[2]: *** Waiting for unfinished jobs....
#14 2386.0 [ 72%] Linking CXX executable gurka_route
#14 2392.9 [ 72%] Built target gurka_route
#14 2398.4 [ 72%] Linking CXX executable gurka_route_with_narrative_languages
#14 2400.5 [ 72%] Built target gurka_route_with_narrative_languages
#14 2400.8 [ 73%] Linking CXX executable gurka_search_side_of_street
#14 2414.6 [ 73%] Built target gurka_search_side_of_street
#14 2426.2 [ 73%] Linking CXX executable gurka_simple_restrictions
#14 2427.9 [ 73%] Linking CXX executable gurka_service_turn
#14 2428.0 [ 73%] Built target gurka_simple_restrictions
#14 2429.3 [ 73%] Built target gurka_service_turn
#14 2430.8 [ 73%] Linking CXX executable gurka_phonemes
#14 2432.0 [ 73%] Built target gurka_phonemes
#14 2432.0 [ 74%] Linking CXX executable gurka_stop_signs
#14 2433.3 [ 74%] Built target gurka_stop_signs
#14 2433.4 make[1]: *** [CMakeFiles/Makefile2:2404: test/CMakeFiles/check.dir/rule] Error 2
#14 2433.6 make: *** [Makefile:735: check] Error 2
```

**Review by kevinkreiser — DISMISSED** [review]
this looks good to me, @monteirodom can you put a changelog entry linking this pr to the bottom of the enhancements list. thanks again!

**Comment by dnesbitt61** [comment]
did a quick comparison using valhalla_run_matrix and it looks good.

**Review by kevinkreiser — APPROVED** [review]

**Comment by kevinkreiser** [comment]
looks like theres a conflict with master, probably in the changelog and i dont have permissions to fix it on your branch. but once you resolve the conflict we can get your PR merged! thanks again for hanging in there!

File: CHANGELOG.md [hunk 1]
```diff
@@ -83,7 +83,8 @@
    * CHANGED: modernized spatialite syntax [#3580](https://github.com/valhalla/valhalla/pull/3580)
    * ADDED: Options to generate partial results for time distance matrix when there is one source (one to many) or one target (many to one). [#3181](https://github.com/valhalla/valhalla/pull/3181)
    * ADDED: Enhance valhalla_build_elevation with LZ4 recompression support [#3607](https://github.com/valhalla/valhalla/pull/3607)
-   * CHANGED: removed UK admin and upgraded its constituents to countries [#3619](https://github.com/valhalla/valhalla/pull/3619)    * CHANGED: expansion service: only track requested max time/distance [#3532](https://github.com/valhalla/valhalla/pull/3509)
+   * ADDED: Shorten down the request delay, when some sources/targets searches are early aborted [#3611](https://github.com/valhalla/valhalla/pull/3611)
```
Comment (kevinkreiser) [inline CHANGELOG.md:86]: can you move this to the end of the list please?

Comment (monteirodom) [reply CHANGELOG.md]: Sure, done.

**Review by kevinkreiser — APPROVED** [review]

**Pull request merged**